### ISSUE TITLE
Login issues and other website problems for c:geo (master ticket)

### ISSUE THREAD

**Issue by moving-bits** [cgeo/cgeo#15555]
This issue serves as a master ticket for current issues for c:geo due to the recent website changes. Bookmark this page for a status overview on progress. Issues marked as "closed" are fixed and will be released with the next nightly version / developer build. We will notify you here when a general public release is published.

Known issues due to recent website changes:
- [x] #15543 - fixed in release 2024.04.05
- [x] #15549 - fixed in release 2024.04.13
- [ ] #15550 / [ ] #15551 - potentially fixed with #15558
- [x] #15562 - fixed with #15569 and again with #15584 - fixed in release 2024.04.13
- [x] #15591 - fixed with #15594 - fixed in release 2024.04.13
- [x] #15633 - fixed in release 2024.04.18, and yet again (after new website changes) in 2024.04.25
- [x] #15638 - fixed in release 2024.04.18
- [ ] #15689 - fixed with #15690 (yet to be released)

**Comment by moving-bits** [comment]
Will start building an interim bugfix release now, including the login fix (but no TB fixes yet, those will take some more time).

- [ ] Check/insert correct version name in `BranchDetectionHelper.java`
- [ ] Wait for a Crowdin cycle or trigger sync manually
- [ ] Merge the related Crowdin translation pull requests for branch `release`
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [ ] Upload and publish ```cgeo-release.apk``` to Google Play
- [ ] Tag the release on Github in format `market_yyyymmdd`
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github
- [ ] Publish release on F-Droid
- [ ] Trigger notification to status.cgeo.org via CI server
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter/X

**Comment by Lineflyer** [comment]
Perfect! Thanks!

**Comment by moving-bits** [comment]
Release 2024.04.05 is rolling, should be available on Play Store in an hour or so, and earlier on F-Droid or via [direct download](https://github.com/cgeo/cgeo/releases/latest).
It includes the login fix (as well as other non-urgent fixes since the last release 2024.03.19).

**Comment by Lineflyer** [comment]
Thanks @moving-bits 

ToDo for this case:
[ ] Check status.cgeo.org and wait until deployment figures are increasing (=Update is live)
[ ] Use CI to trigger update notification to inform users to upgrade
[ ] Use CI to clear custom notification

**Comment by realMrTom** [comment]
Seen update in Playstore and installed it. Now I'm logged in correctly as premium and can see (including logs and fotos), save and log caches. Only finding TB's doesn't work. Thanks for this update. I'm glad for no longer being forced to use the officially app.

**Comment by moving-bits** [comment]
> [ ] Use CI to trigger update notification to inform users to upgrade
> [ ] Use CI to clear custom notification

Please keep a notification until TB handling is fixed as well.

Update is available meanwhile, I'll trigger the default update notification via CI.

**Comment by wolverine007** [comment]
> Seen update in Playstore and installed it. Now I'm logged in correctly as premium and can see (including logs and fotos), save and log caches. Only finding TB's doesn't work. Thanks for this update. I'm glad for no longer being forced to use the officially app.

I needed to refresh the cgeo<>GC.com authentication, kill the app and after a fresh start all was fine.

@moving-bits maybe you could add this to the info box?

**Comment by bekuno** [comment]
The automatic connect to GC was fine after the update for me, no aditional work was necessary.

**Comment by wolverine007** [comment]
> The automatic connect to GC was fine after the update for me, no aditional work was necessary.

Got the same issue from above with two different devices and accounts - login was fine, but Premium was missing

**Comment by Lineflyer** [comment]
Thats another issue we already have listed here. PM is only detected after second login. We should really try to fix it to avoid confusion.

**Comment by Lineflyer** [comment]
Posted on Facebook

**Comment by ztNFny** [comment]
It seems GC changed (reverted) something on their end. Login works fine again with the old version as of right now.

**Comment by eddiemuc** [comment]
> It seems GC changed (reverted) something on their end. Login works fine again with the old version as of right now.

If gc.com reverts it's change then login should still work with my login change. I am unsure about retrieval of "found caches" count though

**Comment by MagpieFourtyTwo** [comment]
@eddiemuc 
> If gc.com reverts it's change then
> login should still work (...)
> retrieval of "found caches" count (...)

Both WFM right now, with latest nightly.

**Comment by technetium** [comment]
I can confirm the app updated on my phone, it's logging in at geocaching.com and I can at l(e)ast post a picture at one of my logs. Thank you for your work and the recent update.

**Comment by itsbrody** [comment]
Since yesterday (afternoon) I can login again; writing (and editing) Logs and sending them. I'm still on build 2024.03.19 and surprised.

**Comment by ztNFny** [comment]
> It seems GC changed (reverted) something on their end. Login works fine again with the old version as of right now.

And broken again 😭 Maybe some A/B testing?

**Comment by MagpieFourtyTwo** [comment]
> @eddiemuc
> 
> > If gc.com reverts it's change then
> > login should still work (...)
> > retrieval of "found caches" count (...)
> 
> Both WFM right now, with latest nightly.

Hrmm - found count was at least strange today.  But perhaps nevertheless correct WRT c:geo's retrieval ...

Scenario: Have visited some caches this afternoon (between about 15:00 and 17:00) and of course saved offline logs, which I now uploaded as drafts to gc.com.  But while writing the first log online, I just noticed that the found counts, c:geo generated with the NUMBER macro this afternoon, where off by 10 caches - which is exactly the number of lab caches I made a few years ago on a MEGA. And which were not counted into the finds count till last week (for whatever reason). But last week I noticed that my found count suddenly jumped up by 10 - after wondering about the reason for a while I thought that GS may have changed their mind about counting event labs now.  But today the count seems to have dropped again by 10 ... most probably in the same time slot when the "old" login did work again ... haven't checked the minutes, but if just looking at the hours, it may have been the same time slot. Seems as if there is even more going on behind the scenes ... 

Or has anyone else had such experiences?  Which, despite everything, could perhaps nevertheless be an indicator of a non-functioning retrieval ...?

**Comment by MagpieFourtyTwo** [comment]
Ah, just stumbled upon a commit which claims to have fixed found counter problems ... will just have an eye on it.
Although right now it works correct again ... (just tested with a fake log) ... heieiei. How is one supposed to make sense of the confusion, the pros over there are constantly causing ...?

**Comment by technetium** [comment]
Unfortunately, yesterday evening I could not log in. Today still not able to login. 

p.s. Do you like/want this kind of status updates, or are these annoying remarks keeping you from fixing the problem?

**Comment by moving-bits** [comment]
Not being able to log in can have several causes. In most cases c:geo shows you some additional info / an error message. The problem mentioned above showed something like "request verification token not found", whereas something like "captcha required" would be some totally different reason, not directly related to the recent website changes.

**Comment by Lineflyer** [comment]
We still get some (but not many) reports of users not being able to login ("failed to find request verification token") with version 2024.04.05. 
Have seen two users on FB and two users on support.

Not sure whats going on.

**Comment by MagpieFourtyTwo** [comment]
Over here (with latest nightly) it's working - at least most of the times:
- after a fresh restart, *not a single login* succeeded within the usual two to three seconds  :/
- most of the times it takes about 25 seconds
- some take even up to 45 seconds, but are at least successful
- if longer than 45 secs, the hang for... ok, don't know for how long, cause being on tour I usually killed c:geo after 2 minutes ;)

Which leads me to a question: What happens if Live Map is started while the login procedure is still active, but has not yet been successful? Seems as if c:geo caches the member state (as it nevertheless indicates ``Premium``), but how about fetching caches? Will this just be delayed or will PM caches be suppressed till login was successful?

**Comment by itsbrody** [comment]
> We still get some (but not many) reports of users not being able to login (**"failed to find request verification token"**) with version 2024.04.05. Have seen two users on FB and two users on support.
> 
> Not sure whats going on.

whilst with 2024.03.19 version again the above mentioned login error occured, the update to fdroids 2024.04.05-version currently seem to work (as least regards the login problem): I get in an acceptable time of less seconds a "login OK - Premium"

**Comment by Lineflyer** [comment]
How to proceed?
Shall we publish another bugfix release to get trackables working again and potential fix for caches outsise visible area on map or is there work in progress?

**Comment by moving-bits** [comment]
> How to proceed? Shall we publish another bugfix release to get trackables working again and potential fix for caches outsise visible area on map or is there work in progress?

I'll update the fix for the yet-again changed find count today and can prepare a new bugfix release (based on `release` and including that fix) tonight.

**Comment by MagpieFourtyTwo** [comment]
> the yet-again changed find count

Perhaps you can even figure out why the login takes so extremely long (at least 25 to 30 and sometimes even up to 90 secs)?
Seems to be connected to the new token and found count fetching ...

BTW: Is there actually an end in sight or are there still changes resp. back and forth actions every single day?

**Comment by moving-bits** [comment]
> Perhaps you can even figure out why the login takes so extremely long (at least 25 to 30 and sometimes even up to 90 secs)? Seems to be connected to the new token and found count fetching ...

AFAIK there have no new calls to the server been added, so response time should be unchanged by that (and thus only be influenced by network connection and server responsiveness).

> BTW: Is there actually an end in sight or are there still changes resp. back and forth actions every single day?

Seems to have calmed down meanwhile. We still have two failing tests wrt to trackables, but other than that all tests are back to green (and seem to stay at that).

**Comment by MagpieFourtyTwo** [comment]
In web browser pages load within fractions of a second, thus server is not an issue over here. 
Haven't timed the logins before, but would swear they were way faster before, in fact within no time, too. At least I have never seen the "Logging in" message that long before this change rally started.

**Comment by bekuno** [comment]
Since some day we have new test errors at CI, see #15591.
I added it to the list above.

**Comment by moving-bits** [comment]
All tests on CI are back to green meanwhile. Will start preparing a new bugfix release now:

- [ ] Check/insert correct version name in `BranchDetectionHelper.java`
- [ ] Wait for a Crowdin cycle or trigger sync manually
- [ ] Merge the related Crowdin translation pull requests for branch `release`
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [ ] Upload and publish ```cgeo-release.apk``` to Google Play
- [ ] Tag the release on Github in format `market_yyyymmdd`
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github
- [ ] Publish release on F-Droid
- [ ] Trigger notification to status.cgeo.org via CI server
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter/X

**Comment by Lineflyer** [comment]
Added #15633 in initial post.

Need a new release once fix is confirmed and merged.

**Comment by moving-bits** [comment]
I'll start preparing a new bugfix release now:

- [ ] Check/insert correct version name in `BranchDetectionHelper.java`
- [ ] Wait for a Crowdin cycle or trigger sync manually
- [ ] Merge the related Crowdin translation pull requests for branch `release`
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [ ] Upload and publish ```cgeo-release.apk``` to Google Play
- [ ] Tag the release on Github in format `market_yyyymmdd`
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github
- [ ] Publish release on F-Droid
- [ ] Trigger notification to status.cgeo.org via CI server
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter/X

**Comment by Lineflyer** [comment]
Posted on FB.
Posting on X still not working due to 2FA to a mail address, where I do not have access to.

**Comment by moving-bits** [comment]
We have relevant new fixes since our last release (2024.04.18), namely the new TB logging fix and the username detection (among other things), thus I propose to release a new bugfix update before the weekend.

**Comment by MagpieFourtyTwo** [comment]
BTW:  Login is way faster again (since a couple of days), takes usually "just" 5 seconds.  Not as fast as "before", but ok.  Sometimes still "Login failed" appears, but usually repairs itself after another 5 seconds.  And although I had to restart c:geo completely yesterday to get logged in, this happened only once. Thus I would consider login pretty "normal" for now.

**Comment by moving-bits** [comment]
Will start working on a new release now:

- [ ] Check/insert correct version name in `BranchDetectionHelper.java`
- [ ] Wait for a Crowdin cycle or trigger sync manually
- [ ] Merge the related Crowdin translation pull requests for branch `release`
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [ ] Upload and publish ```cgeo-release.apk``` to Google Play
- [ ] Tag the release on Github in format `market_yyyymmdd`
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github
- [ ] Publish release on F-Droid
- [ ] Trigger notification to status.cgeo.org via CI server
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter/X

**Comment by moving-bits** [comment]
Will start working on a new release now:

- [ ] Check/insert correct version name in `BranchDetectionHelper.java`
- [ ] Wait for a Crowdin cycle or trigger sync manually
- [ ] Merge the related Crowdin translation pull requests for branch `release`
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [ ] Upload and publish ```cgeo-release.apk``` to Google Play
- [ ] Tag the release on Github in format `market_yyyymmdd`
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github
- [ ] Publish release on F-Droid
- [ ] Trigger notification to status.cgeo.org via CI server
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter/X

**Comment by moving-bits** [comment]
Having merged the open PR I'll resume working on a new bufix release:

Will start working on a new release now:

- [ ] Check/insert correct version name in `BranchDetectionHelper.java`
- [ ] Wait for a Crowdin cycle or trigger sync manually
- [ ] Merge the related Crowdin translation pull requests for branch `release`
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [ ] Upload and publish ```cgeo-release.apk``` to Google Play
- [ ] Tag the release on Github in format `market_yyyymmdd`
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github
- [ ] Publish release on F-Droid
- [ ] Trigger notification to status.cgeo.org via CI server
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter/X

**Comment by fm-sys** [comment]
Should be fixed. Please open follow-up issues if necessary...

**Comment by countdanilo** [comment]
Sadly login still doesn't work for me and hasn't for absolute ages. On v. 2024.07.07. I'm really frustrated here and considering giving c:geo up for good...

**Comment by Narrat** [comment]
Sometimes users needed to solve a captcha in the past, which caused login failures within cgeo.
There is a page on the website
https://manual.cgeo.org/en/loginproblems
Did you check those cases?

**Comment by ztNFny** [comment]
> Sadly login still doesn't work for me and hasn't for absolute ages. 
> On a OnePlus 11 5G running Android 14.0.  c:geo is v. 2024.07.07. 
> I'm really frustrated here and considering giving c:geo up for good... 

At this point there are no known problems, all adjustments to the changed login process were done long ago. Please open a new ticket and provide logs to investigate your issue.
